pkg/resourceutils/listResources.go

```diff
@@ -0,0 +1,192 @@
+	storkClient, err := storkops.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	crdList, err := storkClient.ListApplicationRegistrations()
+	if err != nil {
+		return nil, err
+	}
+	resourceReplicaMap := make(map[string]map[metav1.GroupVersionKind]map[string]string)
+	for _, ns := range activationNamespaces {
+		//Initialise namespace -> resource kind maps
+		resourceReplicaMap[ns] = make(map[metav1.GroupVersionKind]map[string]string)
+
+		// 1. ApplicationResources
+		for _, gvk := range GetGVKsRelevantForScaling() {
+			//Initialise namespace/resourceKind -> resources/replicas maps
+			resourceReplicaMap[ns][gvk] = make(map[string]string)
+			resources, _, err := ListResourcesByGVK(gvk, ns, config)
+			if err != nil {
+				return nil, err
+			}
+			if resources == nil {
+				continue
+			}
+			for _, object := range resources.Items {
+				content := object.UnstructuredContent()
+				replicaSetGVK := metav1.GroupVersionKind{Group: "apps", Version: "v1", Kind: "ReplicaSet"}
+				if gvk == replicaSetGVK {
+					ownerReferences, _, err := unstructured.NestedSlice(content, "metadata", "ownerReferences")
+					if err != nil {
+						return nil, fmt.Errorf("error getting the ownerReferences for the replicaSet: %v/%v -> %v", object.GetNamespace(), object.GetName(), err)
+					}
+					if ownerReferences != nil {
+						// only collect replicaSets without owners
+						continue
+					}
+				}
+				annotations, found, err := unstructured.NestedStringMap(content, "metadata", "annotations")
+				if err != nil {
+					return nil, fmt.Errorf("error getting the annotations of the resource %v %v/%v: %v", strings.ToLower(gvk.Kind), object.GetNamespace(), object.GetName(), err)
+				}
+				if !found {
+					continue
+				}
+				// only if replica annotation exists, we will be activating this resource
+				if val, present := annotations[migration.StorkMigrationReplicasAnnotation]; present {
+					resourceReplicaMap[ns][gvk][object.GetName()] = val
+				}
+			}
+		}
+
+		// 2. CronJobs
+		resourceReplicaMap[ns][cronJobGVK] = make(map[string]string)
+		resources, _, err := ListResourcesByGVK(cronJobGVK, ns, config)
+		if err != nil {
+			return nil, err
+		}
+		if resources != nil {
+			for _, object := range resources.Items {
+				content := object.UnstructuredContent()
+				annotations, found, err := unstructured.NestedStringMap(content, "metadata", "annotations")
+				if err != nil {
+					return nil, fmt.Errorf("error getting the annotations of the resource %v %v/%v: %v", strings.ToLower(cronJobGVK.Kind), object.GetNamespace(), object.GetName(), err)
+				}
+				if !found {
+					continue
+				}
+				// pick only if stork-migrated annotation exists in the resource
+				if val, present := annotations[migration.StorkMigrationAnnotation]; present {
+					resourceReplicaMap[ns][cronJobGVK][object.GetName()] = val
+				}
+			}
```

Comment:
Did we need to create another map here? Couldn't we use the map we have creared in Step1?

Reply:
The resource replica map is a 3 level map
First Key is Namespace
Second Key is the gvk
Third Key is the resource name
And finally the value is the replica value we will set on activation.
Though the value is not required in our current scenario for scaling down, I created the method like this because then it can be plugged in into failover destination activation dry run.

Coming to your point, the map creation is required because resourceReplicaMap[ns][cronJobGVK] is cronJob specific map and we will populate it with the cronJob resources and their new replica values in this particular namespace.